packages/atomic/src/components/facets-v1/atomic-facet/atomic-facet.tsx

```diff
@@ -184,43 +188,85 @@ export class AtomicFacet
   }
 
   private renderValue(facetValue: FacetValue, onClick: () => void) {
+    const displayValue = this.bindings.i18n.t(facetValue.value);
+    const isSelected = facetValue.state === 'selected';
     switch (this.displayValuesAs) {
       case 'checkbox':
         return (
           <FacetValueCheckbox
-            value={facetValue.value}
+            displayValue={displayValue}
             numberOfResults={facetValue.numberOfResults}
-            state={facetValue.state}
+            isSelected={isSelected}
             i18n={this.bindings.i18n}
             onClick={onClick}
           ></FacetValueCheckbox>
         );
       case 'link':
-        return <FacetValueLink value={facetValue.value}></FacetValueLink>;
+        return (
+          <FacetValueLink
+            displayValue={displayValue}
+            numberOfResults={facetValue.numberOfResults}
+            isSelected={isSelected}
+            i18n={this.bindings.i18n}
+            onClick={() => {
+              this.deselectValues();
+              onClick();
+            }}
+          ></FacetValueLink>
+        );
       case 'box':
-        return <FacetValueBox value={facetValue.value}></FacetValueBox>;
+        return (
+          <FacetValueBox
+            displayValue={displayValue}
+            numberOfResults={facetValue.numberOfResults}
+            isSelected={isSelected}
+            i18n={this.bindings.i18n}
+            onClick={onClick}
+          ></FacetValueBox>
+        );
     }
   }
 
-  private renderValues() {
+  private deselectValues() {
+    const action = loadFacetSetActions(this.bindings.engine)
+      .toggleSelectFacetValue;
```

Comment:
the `deselectAllFacetValues` clears the value, and it's not possible to select a value that doesn't exist.
We also have to allow deselecting the value, that wouldn't be possible using toggle after deselecting everything

Reply:
What if we adjust `handleFacetDeselectAll` to set all values to idle instead of setting `currentValues` to an empty array? It seems to me that this would make the `deselectAllFacetValues` action more flexible,

Reply:
handleFacetDeselectAll removes the value to allow the api to "replenish them" when you clear the facet, using breadcrumbs or the Clear Filter button, that shouldn't change, and then you still have the toggle issue

I think we might need new actions to handle this, I've created a new story for it https://coveord.atlassian.net/browse/KIT-729

Reply:
> handleFacetDeselectAll removes the value to allow the api to "replenish them"

Looking back at my notes on the [original PR](https://github.com/coveo/ui-kit/issues/141#issuecomment-733005671), I am pretty sure I cleared the array just because it was simpler. The intent was not to replenish values per say. I also noticed I could not use this approach with the range facets because it would discard manually specified ranges.

Since then, we've talked about the specific facet one day supporting manual ranges (custom order). I think it's inevitable that the deselect logic will need to change, and become closer to the range facets.

Even with the change, I think the api can still replenish values? As long as `freezeCurrentValues` is false, the api can suggest different values even if the array is not empty? Something to validate ...

> and then you still have the toggle issue

Could you give an example of the issue?